### PR TITLE
feat(Tree): expand the tree to see active node(s) on mount

### PR DESCRIPTION
**Pull request by Kenny806**

### CODE REVIEW

File: src/components/Tree/Node.tsx [hunk 1]
```diff
@@ -53,6 +53,31 @@ type NodeProps = {
     onEditableSave?: (targetItemId: string, value: string) => void;
 };
 
+const setShowNodesInitial = (currentNode: DraggableItem<TreeNodeItem>, activeNodeIds?: NullableString[]): boolean => {
```
Comment (imoutaharik) [inline src/components/Tree/Node.tsx:56]: Would it be possible to find another name for this function? I think maybe en `setShowNodesOnInit` would be more semantic.

File: src/components/Tree/Node.tsx [hunk 2]
```diff
@@ -53,6 +53,31 @@ type NodeProps = {
...
+        if (node.nodes) {
+            for (const childNode of node.nodes) {
+                if (activeIds.includes(childNode.id)) {
+                    return true;
+                }
+                if (hasActiveChildNodes(childNode, activeIds)) {
+                    return true;
+                }
+            }
+        }
```
Comment (imoutaharik) [inline src/components/Tree/Node.tsx:67]: Also it would be great to simplify this function.

**Review by imoutaharik — CHANGES_REQUESTED** [review]
Thanks for making the changes! 😄 I think it would be great to add stories when it's collapsed and when it's expanded on load. And a test. 😜 WDYT?

File: src/components/Tree/Node.tsx (hunk 1 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx:56]: What about `getShowNodesOnIt`? Now that I look at it, this function doesn't set a value, but gets one.

Comment (Kenny806) [reply src/components/Tree/Node.tsx]: Renamed to `getInitialShowNodesValue`, as I realized it's not a setter, but rather a getter. And it seems descriptive, as it is getting the initial value for the `showNodes` state variable. But happy to rename to anything else

File: src/components/Tree/Node.tsx (hunk 2 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx:67]: Simplified it :-)

File: src/components/Tree/Tree.cy.tsx [hunk 3]
```diff
@@ -7,11 +7,12 @@ import { Tree, TreeFlatListItem } from './Tree';
 
 type ComponentProps = {
     nodes: DraggableItem<TreeFlatListItem>[];
+    activeNodeIds?: NullableString[];
     onDrop?: (sourceItemId: string, parentId: NullableString, positionBeforeId: NullableString) => void;
     onEditableSave?: (itemId: string, value: string) => void;
 };
-const Component: FC<ComponentProps> = ({ nodes, onDrop, onEditableSave }) => {
-    const [selectedIds, setSelectedIds] = useState<NullableString[]>([]);
+const Component: FC<ComponentProps> = ({ nodes, onDrop, onEditableSave, activeNodeIds }) => {
```
Comment (imoutaharik) [inline src/components/Tree/Tree.cy.tsx:14]: That would be great to emove `FC` from here too.

Comment (Kenny806) [reply src/components/Tree/Tree.cy.tsx]: Done :-)

**Review by imoutaharik — APPROVED** [review]

**Pull request merged**